src/Data/LargeHashable/LargeWord.hs

```diff
@@ -0,0 +1,83 @@
+{-# LANGUAGE DeriveDataTypeable #-}
+{-# LANGUAGE DeriveGeneric #-}
+module Data.LargeHashable.LargeWord
+    ( Word128(..), Word256(..)
+    , fromWord8, toWord8
+    , bsToW64, w64ToBs
+    , bsToW128, w128ToBs
+    , bsToW256, w256ToBs
+    , xorW128, xorW256
+    )
+where
+
+-- keep imports in alphabetic order (in Emacs, use "M-x sort-lines")
+import Data.Bits
+import Data.Data (Data)
+import Data.Typeable
+import Data.Word
+import GHC.Generics (Generic)
+import qualified Data.ByteString as BS
+
+data Word128
+    = Word128
+    { w128_first :: !Word64
+    , w128_second :: !Word64
+    }
+    deriving (Show, Read, Eq, Ord, Typeable, Generic, Data)
+
+data Word256
+    = Word256
+    { w256_first :: !Word128
+    , w256_second :: !Word128
+    }
+    deriving (Show, Read, Eq, Ord, Typeable, Generic, Data)
+
+fromWord8 :: Integral a => Word8 -> a
+fromWord8 = fromIntegral
+
+toWord8 :: (Show a, Integral a) => a -> Word8
+toWord8 i
+    | i > 255 || i < 0 = error ("toWord8: not in range " ++ show i)
+    | otherwise = fromIntegral i
+
+bsToW256 :: BS.ByteString -> Word256
```

Comment:
maybe add a comment that if the bytestring is larger than w256 the rest will be cut off/ignored?